src/js/components/stock-movement-wizard/combined-shipments/AddItemsPage.jsx

```diff
@@ -0,0 +1,874 @@
+        }) => ({
+          onClick: () => {
+            updateTotalCount(1);
+            addRow({
+              product: {
+                ...fieldValue.product,
+                label: `${fieldValue.product.productCode} ${fieldValue.product.name}`,
+              },
+              recipient: fieldValue.recipient,
+              sortOrder: fieldValue.sortOrder + 1,
+              orderItemId: fieldValue.orderItemId,
+              referenceId: fieldValue.id,
+              orderNumber: fieldValue.orderNumber,
+              packSize: fieldValue.packSize,
+            }, rowIndex);
+          },
+        }),
+        attributes: {
+          className: 'btn btn-outline-success',
+        },
+      },
+      deleteButton: DELETE_BUTTON_FIELD,
+    },
+  },
+};
+
+/* eslint class-methods-use-this: ["error",{ "exceptMethods": ["getLineItemsToBeSaved"] }] */
+/**
+ * The second step of stock movement where user can add items to stock list.
+ * This component supports three different cases: with or without stocklist
+ * when movement is from a depot and when movement is from a vendor.
+ */
+class AddItemsPage extends Component {
+  static updateSortOrder(lineItems) {
+    return _.map(lineItems, (item, rowIndex) => ({
+      ...item,
+      sortOrder: (item.sortOrder - (item.sortOrder % 100)) + rowIndex + 1,
+    }));
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      values: { ...this.props.initialValues, lineItems: [] },
+      totalCount: 0,
+      isFirstPageLoaded: false,
+    };
+
+    this.props.showSpinner();
+    this.confirmSave = this.confirmSave.bind(this);
+    this.validate = this.validate.bind(this);
+    this.isRowLoaded = this.isRowLoaded.bind(this);
+    this.loadMoreRows = this.loadMoreRows.bind(this);
+    this.updateTotalCount = this.updateTotalCount.bind(this);
+    this.updateRow = this.updateRow.bind(this);
+    this.removeItem = this.removeItem.bind(this);
+    this.fetchLineItems = this.fetchLineItems.bind(this);
+
+    this.debouncedProductsFetch = debounceProductsFetch(
+      this.props.debounceTime,
+      this.props.minSearchLength,
+      this.props.initialValues.origin.id,
+    );
+  }
+
+  componentDidMount() {
+    if (this.props.stockMovementTranslationsFetched) {
+      this.dataFetched = true;
+
+      this.fetchAllData(false);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.stockMovementTranslationsFetched && !this.dataFetched) {
+      this.dataFetched = true;
+
+      this.fetchAllData(false);
+    }
+  }
+
+  /**
+   * Returns an array of new stock movement's items and items to be
+   * updated (comparing to previous state of line items).
+   * @param {object} lineItems
+   * @public
+   */
+  getLineItemsToBeSaved(lineItems) {
+    const items = AddItemsPage.updateSortOrder(lineItems);
+    const lineItemsToBeAdded = _.filter(items, item => !item.id);
+    const lineItemsToBeUpdated = _.filter(items, item => item.id);
+
+    return [].concat(
```

Comment:
Is there a need to do the two separate maps and concat them? Looks like the only difference here is the id field. I think this could be sent as `null` if there is no item.id (`id: item.id ? item.id : null`).